### DISCUSSION TITLE
Using FullPageOS as a makeshift Online Audio Player

### DISCUSSION THREAD

**Discussion by SecularSteve** [guysoft/FullPageOS#514] [General]
The following YouTube API Web-based Open Source Clients are handy for that. Though they all come with their own issues:

-----

**ytify**
https://github.com/n-ce/ytify
https://ytify.netlify.app/

Minimalist, lightweight, fast, compact, configurable and above all feature-packed.
It's like it was meant for this system.
Wouldn't recommend it if you're looking to play Live Streams.
It's still being worked on.

-----

**Beatbump**
https://github.com/snuffyDev/Beatbump
https://beatbump.io/home

Possible the one most YT-Music like in this list. 
It has the least amount of themes, the Video function is pretty experimental, and sometimes it'll pick the video links instead of the Topic links regardless of how you click on the Album.

-----

**HyperPipe**
https://codeberg.org/Hyperpipe/Hyperpipe
https://hyperpipe.surge.sh/

The one with the most potential, but the least stable one.
Unlike ytify, it tries to be less lightweight with it's resources, and it's solely focused on YT Music's Topic Channels (even though it can play videos).
Use this one if you just want the experimental basics.

-----

**FreeTube Cordova**
https://github.com/MarmadileManteater/FreeTubeCordova
https://marmadilemanteater.github.io/freetube/#/

Basically, a PWA (Progressive Web App) configured fork of the well-known desktop-based FreeTube client.
Still somewhat experimental, but I have little to no issues with using it. It's ideal if you want a more advanced, but resource-heavy version of ytify. Not really ideal for audio-playback (has some similar issues to Beatbump), but it's a slight bit more stable, so that's a plus.

**Comment by guysoft** [comment]
Did you manage to get any of them to work?
